servers/rendering/shader_types.cpp

```diff
@@ -516,6 +516,7 @@ ShaderTypes::ShaderTypes() {
 	shader_modes[RS::SHADER_FOG].functions["fog"].built_ins["EMISSION"] = ShaderLanguage::TYPE_VEC3;
 	shader_modes[RS::SHADER_FOG].functions["fog"].main_function = true;
 
+	// Must be kept in sync with the Shader::Mode enum.
```

Comment:
Can we use a `static_assert()` or similar to guarantee sizes match at compile-time? I'm not sure if this is possible.

Reply:
Can't use `static_assert`, but I've updated the PR to add `DEV_ASSERT`. Note that this requires including the `shader.h` header.